src/config_generator/from_proto.rs

```diff
@@ -0,0 +1,387 @@
+#![allow(unused)]
+
+use std::collections::{BTreeSet, HashMap};
+
+use convert_case::{Case, Casing};
+use derive_setters::Setters;
+use prost_reflect::prost_types::{
+    DescriptorProto, EnumDescriptorProto, FileDescriptorSet, ServiceDescriptorProto,
+};
+use strum_macros::Display;
+
+use crate::blueprint::GrpcMethod;
+use crate::config::{Arg, Config, Field, Grpc, Tag, Type};
+
+pub(super) static DEFAULT_SPECTATOR: &str = "_";
+
+/// Enum to represent the type of the descriptor
+#[derive(Display, Clone)]
+pub enum DescriptorType {
+    Enum,
+    Message,
+    Query,
+}
+
+#[derive(Default)]
+struct Helper {
+    map: HashMap<String, String>,
+    package: String,
+}
```

Comment:
Do we need this type?

Reply:
It's the backbone. It provides insert and get function to easily deal with character cases